models/league.py

```diff
@@ -25,8 +25,55 @@ class League:
         - len(self.matches) > 0
     """
 
-    teams: dict[str, Team]
-    matches: list[Match]
+    _teams: dict[str, Team]
+    _matches: list[Match]
+
+    def __init__(self) -> None:
+        self._teams = {}
+        self._matches = []
+
+    def add_team(self, name: str) -> Team:
+        """Add a new team with the given team name to this league and return it.
+
+        Preconditions
+            - name not in self._teams
+        """
+        team = Team(name=name, matches=[])
+        self._teams[name] = team
+        return team
+
+    def add_match(self, team1: str, team2: str, match: Match) -> None:
```

Comment:
Since _matches is chronologically ordered, we should probably either add a precondition to add_match to only allow inserting the latest match or insert the match in the right position in matches.

Reply:
I tried adding the precondition, check it over